website/versioned_docs/version-experimental/RelayHooks-StepByStep.md

```diff
@@ -0,0 +1,300 @@
+
+```javascript
+// your-app-name/src/fetchGraphQL.js
+async function fetchGraphQL(text, variables) {
+  const REACT_APP_GITHUB_AUTH_TOKEN = process.env.REACT_APP_GITHUB_AUTH_TOKEN;
+
+  // Fetch data from GitHub's GraphQL API:
+  const response = await fetch('https://api.github.com/graphql', {
+    method: 'POST',
+    headers: {
+      Authorization: `bearer ${REACT_APP_GITHUB_AUTH_TOKEN}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      query: text,
+      variables,
+    }),
+  });
+
+  // Get the response as JSON
+  return await response.json();
+}
+
+export default fetchGraphQL;
+```
+
+### 2.3: Fetching GraphQL From React
+
+Now we can use our `fetchGraphQL` function to fetch some data in our app. Open `src/App.js` and edit it as follows:
+
+```javascript
+// your-app-name/src/App.js
+import React from 'react';
+import './App.css';
+import fetchGraphQL from './fetchGraphQL';
+
+const { useState, useEffect } = React;
+
+function App() {
+  // We'll load the name of a repository, initially setting it to null
+  const [name, setName] = useState(null);
+
+  // When the component mounts we'll fetch a repository name
+  useEffect(() => {
+    let isMounted = true;
+    fetchGraphQL(`
+      query RepositoryNameQuery {
+        # feel free to change owner/name here
+        repository(owner: "facebook" name: "relay") {
+          name
+        }
+      }
+    `).then(response => {
+      // Avoid updating state if the component unmounted before the fetch completes
+      if (!isMounted) {
+        return;
+      }
+      const data = response.data;
+      setName(data.repository.name);
+    }).catch(error => {
+      console.error(error);
+    });
+
+    return () => {
+      isMounted = false;
+    };
+  }, [fetchGraphQL]);
+
+  // Render "Loading" until the query completes
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>
+          {name != null ? `Repository: ${name}` : "Loading"}
+        </p>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+## Step 3: When To Use Relay
+
+At this point we can fetch data with GraphQL and render it with React. This is a reasonable solution that can be sufficient for many apps. However, this approach doesn't necessarily scale. As our app grows in size and complexity, or the number of people working on the app grows, a simple approach like this can become limiting. Relay provides a number of features designed to help keep applications fast and reliable even as they grow in size and complexity: colocating data dependencies in components with GraphQL fragments, data consistency, mutations, etc. Check out [Thinking in GraphQL](/docs/en/thinking-in-graphql.html) and [Thinking in Relay](/docs/en/thinking-in-relay.html) for an overview of how Relay makes it easier to work with GraphQL.
```

Comment:
not a blocker, but might be good to expand on this section and possibly show people the limitations they'd run into with concrete examples